partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/TypeEnv.kt

```diff
@@ -22,10 +23,16 @@ import org.partiql.value.stringValue
  * @property outer refers to the outer variable scopes that we have access to.
  */
 internal data class TypeEnv(
+    private val globals: Env,
     public val schema: List<Rel.Binding>,
     public val outer: List<TypeEnv>
 ) {
 
+    enum class LookupStrategy {
+        LOCALS_FIRST,
+        GLOBALS_FIRST
+    }
```

Comment:
Perhaps we rename Scope.kt to Strategy.kt per https://github.com/partiql/partiql-lang-kotlin/blob/v1-planner/partiql-planner/src/main/kotlin/org/partiql/planner/intern/validate/Strategy.kt